src/Indicator.vala

```diff
@@ -161,6 +169,18 @@ public class Session.Indicator : Wingpanel.Indicator {
     }
 
     public override void closed () {}
+
+    private void show_shutdown_dialog (uint32 timestamp = 0) {
+        if (shutdown_dialog != null) {
+            shutdown_dialog.present ();
+            return;
+        }
+
```

Comment:
@donadigo @davidmhewitt I would just flip it. and remove the time stamp argument since you're not using it
```
    if (shutdown_dialog == null) {
        shutdown_dialog = new Session.Widgets.EndSessionDialog (Session.Widgets.EndSessionDialogType.RESTART);
        shutdown_dialog.destroy.connect (() => { shutdown_dialog = null; });
        shutdown_dialog.set_transient_for (indicator_icon.get_toplevel () as Gtk.Window);  
      }
 
      
      shutdown_dialog.present ();
      shutdown_dialog.show_all ();
  }```

Or better yet, why not just always create it when you use it, and not keep it in memory? It's a simple enough dialog that creating it on the fly shouldn't be a problem :wink: